integrations/integrations.php

```diff
@@ -13,49 +18,72 @@ class Integrations {
 	/**
 	 * Collection of registered integrations.
 	 *
-	 * @var array<Integration>
+	 * @var array<string,Integration>
 	 */
 	private array $integrations = [];
 
 	/**
 	 * Registers an integration.
 	 *
-	 * @param string             $slug            A unique identifier for the integration.
-	 * @param string|Integration $class_or_object Fully-qualified class or instantiated Integration object.
+	 * @param Integration $integration Instantiated integration object.
+	 *
+	 * @throws InvalidArgumentException Exception if invalid argument are passed.
 	 *
 	 * @private
 	 */
-	public function register( string $slug, $class_or_object ): void {
-		if ( isset( $this->integrations[ $slug ] ) ) {
+	public function register( $integration ): void {
+		if ( ! is_subclass_of( $integration, Integration::class ) ) {
+			throw new InvalidArgumentException( sprintf( 'Integration class "%s" must extend %s.', get_class( $integration ), Integration::class ) );
+		}
+
+		$slug = $integration->get_slug();
+
+		if ( null !== $this->get_integration( $slug ) ) {
 			throw new InvalidArgumentException( sprintf( 'Integration with slug "%s" is already registered.', $slug ) );
 		}
 
-		if ( is_object( $class_or_object ) ) {
-			$integration_object = $class_or_object;
-		} else {
-			if ( ! class_exists( $class_or_object ) ) {
-				throw new InvalidArgumentException( sprintf( 'Integration class "%s" does not exist.', $class_or_object ) );
+		$this->integrations[ $slug ] = $integration;
+	}
+
+	/**
+	 * Returns a registered integration for a key, or null if not found.
+	 *
+	 * @param string $slug A unique identifier for the integration.
+	 */
+	private function get_integration( string $slug ): ?Integration {
+		return $this->integrations[ $slug ] ?? null;
+	}
+
+	/**
+	 * Activates integrations based on the configuration provided by VIP
+	 * (only if not already activated by customer).
+	 *
+	 * @return void
+	 */
+	public function activate_integrations_via_vip_config() {
+		foreach ( $this->integrations as $slug => $integration ) {
+			// Don't activate again if integration is already activated and configured by customer.
+			if ( $integration->is_active() ) {
+				continue;
 			}
 
-			$integration_object = new $class_or_object();
-		}
+			$vip_config = $this->get_integration_config( $slug );
 
-		if ( ! is_subclass_of( $integration_object, Integration::class ) ) {
-			throw new InvalidArgumentException( sprintf( 'Integration class "%s" must extend %s.', get_class( $integration_object ), Integration::class ) );
+			if ( $vip_config->is_active_via_vip() ) {
+				$this->activate( $slug, $vip_config->get_site_config() );
```

Comment:
Here's an idea: let's change the param name for `Integration::activate` and `Integrations::activate` to `options` and `config` becomes one of the supported keys of that.

```
public function activate( string $slug, array $options = [] ): void {
```

`$options` would accept common configuration parameters across all integrations (e.g. `version`) and `$options['config']` would be the integration-specific configs.

```suggestion
				$this->activate( $slug, [
					'config' => $vip_config->get_site_config(),
				] );
```

In the future, we could even do something like this (where `get_integration_opts()` could return those generic configs stored at the platform-level):

```
				$this->activate( $slug, array_merge( $vip_config->get_integration_opts(), [
					'config' => $vip_config->get_site_config(),
				] ) );
```

@alecgeatches curious to hear your perspective on this as well.

Reply:
Feedback implemented in https://github.com/Automattic/vip-go-mu-plugins/pull/4660/commits/2052aadf18938305de1f026468984550bbf96778